internal/driver/gl/window.go

```diff
@@ -628,14 +631,25 @@ func (w *window) mouseClicked(viewport *glfw.Window, button glfw.MouseButton, ac
 	ev := new(fyne.PointEvent)
 	ev.Position = fyne.NewPos(x, y)
 
+	// Prevent async mouse functions from arriving out of order
+	var mouseWG sync.WaitGroup
```

Comment:
This is about tapping or double tapping - a more descriptive name would be helpful

Reply:
This is a general bug fix - without the wait group there is no guarantee as to the order of event arrival.

Reply:
How about eventOrder? The IDE clearly knows this is. WaitGroup so we can remove “WG”. (I’m aware that this naming has been used before but I’ll try to fix that).
One thought though - is this a separate fix, and did we not consider a bigger fix that will resolve keyboard triggers as well? This could apply to all input events not just mouse/tap.

Reply:
Will this work "most of the time" without the waitgroup? we need to find the balance between not blocking and ensuring event order - yes that in itself is probably a bigger issue.

Reply:
Yes. I'll revert it for the time being. Selection behaviour seems to be pretty resilient now to badly ordered events.